packages/tao/src/shared/workspace.ts

```diff
@@ -105,6 +105,21 @@ export interface ProjectConfiguration {
   generators?: { [collectionName: string]: { [generatorName: string]: any } };
 }
 
+export interface TargetDependencyConfig {
```

Comment:
I tihnk we should also add 'configuration'?

For instance, if target "deploy" wants to invoke prod build instead of regular build. 

Maybe we can inherit the config of the running target but you can override it?